diff -uNr orig/arch/arm64/boot/dts/rockchip/px30.dtsi patched/arch/arm64/boot/dts/rockchip/px30.dtsi
--- orig/arch/arm64/boot/dts/rockchip/px30.dtsi	2023-06-09 09:58:41.967826296 +0300
+++ patched/arch/arm64/boot/dts/rockchip/px30.dtsi	2023-06-09 09:56:06.389334279 +0300
@@ -29,9 +29,6 @@
 		i2c1 = &i2c1;
 		i2c2 = &i2c2;
 		i2c3 = &i2c3;
-		mmc0 = &sdmmc;
-		mmc1 = &sdio;
-		mmc2 = &emmc;
 		serial0 = &uart0;
 		serial1 = &uart1;
 		serial2 = &uart2;
@@ -414,24 +411,6 @@
 			compatible = "linaro,optee-tz";
 			method = "smc";
 		};
-
-		scmi: scmi {
-			compatible = "arm,scmi-smc";
-			shmem = <&scmi_shmem>;
-			arm,smc-id = <0x82000010>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			scmi_clk: protocol@14 {
-				reg = <0x14>;
-				#clock-cells = <1>;
-			};
-		};
-
-		sdei: sdei {
-			compatible = "arm,sdei-1.0";
-			method = "smc";
-		};
 	};
 
 	gmac_clkin: external-gmac-clock {
@@ -539,11 +518,6 @@
 		clock-output-names = "xin32k";
 	};
 
-	scmi_shmem: scmi-shmem@10f000 {
-		compatible = "arm,scmi-shmem";
-		reg = <0x0 0x0010f000 0x0 0x100>;
-	};
-
 	pmu: power-management@ff000000 {
 		compatible = "rockchip,px30-pmu", "syscon", "simple-mfd";
 		reg = <0x0 0xff000000 0x0 0x1000>;
@@ -672,10 +646,9 @@
 		interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&pmucru SCLK_UART0_PMU>, <&pmucru PCLK_UART0_PMU>;
 		clock-names = "baudclk", "apb_pclk";
-		dmas = <&dmac 0>, <&dmac 1>;
-		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
+	//	dmas = <&dmac 0>, <&dmac 1>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&uart0_xfer &uart0_cts &uart0_rts>;
 		status = "disabled";
@@ -875,10 +848,9 @@
 		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru SCLK_UART1>, <&cru PCLK_UART1>;
 		clock-names = "baudclk", "apb_pclk";
-		dmas = <&dmac 2>, <&dmac 3>;
-		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
+		dmas = <&dmac 2>, <&dmac 3>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&uart1_xfer &uart1_cts &uart1_rts>;
 		status = "disabled";
@@ -890,12 +862,11 @@
 		interrupts = <GIC_SPI 17 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru SCLK_UART2>, <&cru PCLK_UART2>;
 		clock-names = "baudclk", "apb_pclk";
-		dmas = <&dmac 4>, <&dmac 5>;
-		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
+		dmas = <&dmac 4>, <&dmac 5>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&uart2m0_xfer>;
+		pinctrl-0 = <&uart2m1_xfer>;
 		status = "disabled";
 	};
 
@@ -905,12 +876,11 @@
 		interrupts = <GIC_SPI 18 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru SCLK_UART3>, <&cru PCLK_UART3>;
 		clock-names = "baudclk", "apb_pclk";
-		dmas = <&dmac 6>, <&dmac 7>;
-		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
+	//	dmas = <&dmac 6>, <&dmac 7>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&uart3m1_xfer &uart3m1_cts &uart3m1_rts>;
+		pinctrl-0 = <&uart3m0_xfer >;
 		status = "disabled";
 	};
 
@@ -920,12 +890,11 @@
 		interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru SCLK_UART4>, <&cru PCLK_UART4>;
 		clock-names = "baudclk", "apb_pclk";
-		dmas = <&dmac 8>, <&dmac 9>;
-		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
+		dmas = <&dmac 8>, <&dmac 9>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&uart4_xfer &uart4_cts &uart4_rts>;
+		pinctrl-0 = <&uart4_xfer>;
 		status = "disabled";
 	};
 
@@ -935,17 +904,16 @@
 		interrupts = <GIC_SPI 20 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru SCLK_UART5>, <&cru PCLK_UART5>;
 		clock-names = "baudclk", "apb_pclk";
-		dmas = <&dmac 10>, <&dmac 11>;
-		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
+//		dmas = <&dmac 10>, <&dmac 11>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&uart5_xfer &uart5_cts &uart5_rts>;
 		status = "disabled";
 	};
 
 	i2c0: i2c@ff180000 {
-		compatible = "rockchip,px30-i2c", "rockchip,rk3399-i2c";
+		compatible = "rockchip,rk3399-i2c";
 		reg = <0x0 0xff180000 0x0 0x1000>;
 		clocks =  <&cru SCLK_I2C0>, <&cru PCLK_I2C0>;
 		clock-names = "i2c", "pclk";
@@ -958,7 +926,7 @@
 	};
 
 	i2c1: i2c@ff190000 {
-		compatible = "rockchip,px30-i2c", "rockchip,rk3399-i2c";
+		compatible = "rockchip,rk3399-i2c";
 		reg = <0x0 0xff190000 0x0 0x1000>;
 		clocks = <&cru SCLK_I2C1>, <&cru PCLK_I2C1>;
 		clock-names = "i2c", "pclk";
@@ -971,7 +939,7 @@
 	};
 
 	i2c2: i2c@ff1a0000 {
-		compatible = "rockchip,px30-i2c", "rockchip,rk3399-i2c";
+		compatible = "rockchip,rk3399-i2c";
 		reg = <0x0 0xff1a0000 0x0 0x1000>;
 		clocks = <&cru SCLK_I2C2>, <&cru PCLK_I2C2>;
 		clock-names = "i2c", "pclk";
@@ -984,7 +952,7 @@
 	};
 
 	i2c3: i2c@ff1b0000 {
-		compatible = "rockchip,px30-i2c", "rockchip,rk3399-i2c";
+		compatible = "rockchip,rk3399-i2c";
 		reg = <0x0 0xff1b0000 0x0 0x1000>;
 		clocks = <&cru SCLK_I2C3>, <&cru PCLK_I2C3>;
 		clock-names = "i2c", "pclk";
@@ -1004,8 +972,9 @@
 		clock-names = "spiclk", "apb_pclk";
 		dmas = <&dmac 12>, <&dmac 13>;
 		dma-names = "tx", "rx";
-		pinctrl-names = "default";
+		pinctrl-names = "default", "high_speed";
 		pinctrl-0 = <&spi0_clk &spi0_csn &spi0_miso &spi0_mosi>;
+		pinctrl-1 = <&spi0_clk_hs &spi0_csn &spi0_miso_hs &spi0_mosi_hs>;
 		#address-cells = <1>;
 		#size-cells = <0>;
 		status = "disabled";
@@ -1019,8 +988,9 @@
 		clock-names = "spiclk", "apb_pclk";
 		dmas = <&dmac 14>, <&dmac 15>;
 		dma-names = "tx", "rx";
-		pinctrl-names = "default";
+		pinctrl-names = "default", "high_speed";
 		pinctrl-0 = <&spi1_clk &spi1_csn0 &spi1_csn1 &spi1_miso &spi1_mosi>;
+		pinctrl-1 = <&spi1_clk_hs &spi1_csn0 &spi1_csn1 &spi1_miso_hs &spi1_mosi_hs>;
 		#address-cells = <1>;
 		#size-cells = <0>;
 		status = "disabled";
diff -uNr orig/arch/arm64/boot/dts/rockchip/rk3326-linux.dtsi patched/arch/arm64/boot/dts/rockchip/rk3326-linux.dtsi
--- orig/arch/arm64/boot/dts/rockchip/rk3326-linux.dtsi	2023-06-09 09:58:43.501144421 +0300
+++ patched/arch/arm64/boot/dts/rockchip/rk3326-linux.dtsi	2023-06-09 09:53:14.830895889 +0300
@@ -7,14 +7,8 @@
 / {
 	compatible = "rockchip,linux", "rockchip,rk3326";
 
-	aliases {
-		mmc0 = &emmc;
-		mmc1 = &sdmmc;
-		mmc2 = &sdio;
-	};
-
 	chosen {
-		bootargs = "earlycon=uart8250,mmio32,0xff160000 console=ttyFIQ0 rw root=PARTUUID=614e0000-0000 rootwait";
+		bootargs = "earlycon=uart8250,mmio32,0xff160000 console=ttyFIQ0 rw rootwait";
 	};
 
 	fiq-debugger {
@@ -22,11 +16,11 @@
 		rockchip,serial-id = <2>;
 		rockchip,wake-irq = <0>;
 		/* If enable uart uses irq instead of fiq */
-		rockchip,irq-mode-enable = <1>;
+		rockchip,irq-mode-enable = <0>;
 		rockchip,baudrate = <1500000>;  /* Only 115200 and 1500000 */
 		interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_LOW>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&uart2m0_xfer>;
+		pinctrl-0 = <&uart2m1_xfer>;
 		status = "okay";
 	};
 
